subprojects/docs/src/docs/userguide/workingWithDependencies.adoc

```diff
@@ -15,73 +15,47 @@
-        </sample>
-++++
+None of the <<inspecting_dependencies,dependency reporting>> helps you with inspecting or further processing the underlying, resolved artifacts of a module. A typical use case for wanting to access the artifacts is to copy them into a specific directory or filter out files of interest based on a specific file extension.
 
-Sometimes you want the library files of a subset of the configuration dependencies (e.g. of a single dependency).
+You can iterate over the complete set of artifacts resolved for a module with the help of the method api:org.gradle.api.file.FileCollection#getFiles()[]. Using this method on a `Configuration` instance is possible as the interface extends `FileCollection`.
 
 ++++
-<sample id="configurationHandlingFiles" dir="userguide/artifacts/configurationHandling" title="Configuration.files with spec">
-            <sourcefile file="build.gradle" snippet="files"/>
-            <output args="-q files"/>
-        </sample>
+<sample id="iterating-artifacts" dir="userguide/dependencies/iteratingArtifacts" title="Iterating over the artifacts resolved for a module">
+    <sourcefile file="build.gradle" snippet="iteration-task" />
+</sample>
 ++++
 
-The `Configuration.files` method always retrieves all artifacts of the _whole_ configuration. It then filters the retrieved files by specified dependencies. As you can see in the example, transitive dependencies are included.
+[NOTE]
+====
+Iterating over the artifacts of a module automatically resolves the configuration. A resolved configuration becomes immutable and cannot add or remove dependencies. If needed you can copy a configuration for further modification via api:org.gradle.api.artifacts.Configuration#copy()[].
+====
 
-You can also copy a configuration. You can optionally specify that only a subset of dependencies from the original configuration should be copied. The copying methods come in two flavors. The `copy` method copies only the dependencies belonging explicitly to the configuration. The `copyRecursive` method copies all the dependencies, including the dependencies from extended configurations.
+=== Accessing a module's metadata artifact
 
-++++
-<sample id="configurationHandlingCopy" dir="userguide/artifacts/configurationHandling" title="Configuration.copy">
-            <sourcefile file="build.gradle" snippet="copy"/>
-            <output args="-q copy"/>
-        </sample>
-++++
+As part of the dependency resolution process, Gradle downloads the main artifact and the metadata file for a module and stores them in the dependency cache. The initial download requires a active connection to the repositories declared in the build. Some organizations enforce strong restrictions on accessing repositories outside of internal network. Instead of downloading artifacts, those organizations prefer to provide an "installable" Gradle cache with all artifacts contained in it to fulfill the build's dependency requirements.
```

Comment:
:x: "Gradle downloads the main artifact and the metadata file for a module and stores them" -> "Gradle downloads the metadata file of a module and stores it"

(If the "main artifact" is not always downloaded for every dependency. Depends on repository type and the new metadata sources configuration.)

Reply:
The part after the first sentence starting with _The initial download..._  feels unrelated to this section. Which is only about the modification api, no? Maybe there is some better place in the docs for this?

Reply:
> "Gradle downloads the main artifact and the metadata file for a module and stores them" -> "Gradle downloads the metadata file of a module and stores it"

done

> The part after the first sentence starting with The initial download... feels unrelated to this section.

I agree. Works better without. Removed.